### PR TITLE
PLANNER-2047: Using Task Assigning with DDL prevents users from completing tasks

### PR DESCRIPTION
**Pull request by wmedvede**
(cherry picked from commit 19822d8c7b83f7624dcf8563d10192f0057df80c)

### CODE REVIEW

**Comment by wmedvede** [comment]
Hi @rsynek would you mind review this PR?

It's the master version for the RHPAM-3027

It's related with https://github.com/kiegroup/jbpm/pull/1691 but can be merged separately

**Comment by wmedvede** [comment]
jenkins execute full downstream build

File: kie-server-parent/kie-server-services/kie-server-services-task-assigning/kie-server-services-task-assigning-runtime/src/main/java/org/kie/server/services/taskassigning/runtime/command/DelegateAndSaveCommand.java [hunk 1]
```diff
@@ -38,6 +38,9 @@
     static final String TASK_MODIFIED_ERROR_MSG = "Task: %s was modified by an external action since the last executed plan," +
             " current status is %s but the expected should be in %s";
 
+    static final String TASK_MODIFIED_ERROR_MSG_1 = "Task: %s was not found, it might have been deleted by an external action since the last executed plan." +
```
Comment (rsynek) [inline kie-server-parent/kie-server-services/kie-server-services-task-assigning/kie-server-services-task-assigning-runtime/src/main/java/org/kie/server/services/taskassigning/runtime/command/DelegateAndSaveCommand.java:41]: soft suggestion: naming variables or constants should reflect their purpose. Here we use a number just to distinguish two constants from each other.

**Comment by wmedvede** [comment]
Non related tests failed but artifacts were generated well, etc.

--
  | Test Result (4 failures / +4)org.jbpm.prediction.pmml.PMMLLogisticRegressionPredictionServiceProcessTest.testUserJohnorg.jbpm.prediction.pmml.PMMLLogisticRegressionPredictionServiceProcessTest.testUserMaryorg.jbpm.prediction.pmml.PMMLRandomForestPredictionServiceProcessTest.testUserJohnorg.jbpm.prediction.pmml.PMMLRandomForestPredictionServiceProcessTest.testUserMary

**Pull request merged**